test/legacy_test/test_apply.py

```diff
@@ -0,0 +1,81 @@
+#   Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+
+import paddle
+
+
+class TestTensorApplyAPI(unittest.TestCase):
+    def setUp(self):
+        self.x = paddle.to_tensor([1, 2, 3, 4, 5], stop_gradient=True)
```

Comment:
we need test input tensor in both CPU and GPU

Reply:
are we support other data types? eg float16, bfloat16 and float64. if yes, should add more case in here

Reply:
Done add the test case